packages/x-charts-pro/src/ChartZoomSlider/internals/ChartAxisZoomSliderThumb.tsx

```diff
@@ -13,14 +13,12 @@ import {
 
 const Rect = styled('rect')(({ theme }) => ({
   [`&.${chartAxisZoomSliderThumbClasses.root}`]: {
-    fill:
-      theme.palette.mode === 'dark'
-        ? (theme.vars || theme).palette.grey[300]
-        : (theme.vars || theme).palette.common.white,
-    stroke:
-      theme.palette.mode === 'dark'
-        ? (theme.vars || theme).palette.grey[600]
-        : (theme.vars || theme).palette.grey[500],
+    fill: (theme.vars || theme).palette.common.white,
+    stroke: (theme.vars || theme).palette.grey[500],
+    ...theme.applyStyles('dark', {
```

Comment:
mode in `theme.palette,mode` doesn't get updated value when theme changes, when used inside `styled` function (not sure why though), Hence i used `theme.applyStyles` which is more reliable.

After this change, there are no more instances of `theme.palette.mode` inside `styled`